### ISSUE TITLE
FuncUI TextBox.onTextChange issues

### ISSUE THREAD

**Issue by sixilli** [fable-compiler/fable-elmish#282]
### Description
I've described the issue here https://github.com/fsprojects/Avalonia.FuncUI/discussions/389
However I'm facing a new issue. For some reason updating a requests that has another request with the same URL, both URLs get updated when the 2nd request is selected. I'm not really sure what's causing this to happen. 

![image](https://github.com/elmish/elmish/assets/6243871/ed1a852a-77bd-4be3-8978-ca8e8e6cb580)

### Repro code
https://github.com/sixilli/funcman/tree/main/src/App

### Expected and actual results
I expect that only 1 request should get updated and switching between requests should not cause unnecessary updates. 

### Related information 

* elmish version: 4.1
* Operating system: Windows

**Comment by et1975** [comment]
Without looking at the sources this feels like state update is not
serialized. Unlike JS runtime, CLR may trigger message dispatch
concurrently and Avalonia should have provided a synchronization mechanism.
I'd start by checking if instantiating Program is using Avalonia's sync
mechanism correctly.

**Comment by sixilli** [comment]
That would have been a good catch, but sadly the issue still persists. I've now switched `Program.run` to `Program.runWithAvaloniaSyncDispatch()`

**Comment by et1975** [comment]
Not dismissing your problem, but unless you can show that the issue is in this library then you need to take it up with FuncUI to get to resolution.

**Comment by sixilli** [comment]
Yeah I've reached out to them as well but haven't heard yet. The problem is most likely in how TextBox is doing things under the hood.
